File: leetcode/algorithms/520_detect_capital/main.py
class DetectCapital:
    def detectCapitalUse(self, word: str) -> bool:
        return (
            word == word.lower()
            or word == word.upper()
            or word == word[0].upper() + word[1:].lower()
        )


    # Solution
    # Solution 1
    def solution1(self, word: str) -> bool:
        return word.isupper() or word.islower() or word.istitle()

    # Solution 2
    def solution2(self, word: str) -> bool:
        c = 0

        for i in word:
            if i == i.upper():
                c += 1

        return c == len(word) or (c == 1 and word[0] == word[0].upper()) or c == 0

File: leetcode/algorithms/520_detect_capital/main.cpp
#include <string>
using namespace std;

class DetectCapital {
public:
    bool detectCapitalUse(string word) {
        for (int i = 1; i < word.size(); i++) {
            if (i > 1) {
                if (word[i] > 'Z' && word[i - 1] < 'a') {
                    return false;
                } else if (word[i] < 'a' && word[i - 1] > 'Z') {
                    return false;
                }
            } else {
                if (word[i - 1] > 'Z' && word[i] < 'a') {
                    return false;
                }
            }
        }
        return true;
    }

    // Solution
    bool solution(string word) {
        if (word.length() < 2) {
            return true;
        }
        if (isupper(word[0]) && isupper(word[1])) {
            for (int i = 2; i < word.length(); i++) {
                if (islower(word[i])) {
                    return false;
                }
            }
        } else {
            for (int i = 1; i < word.length(); i++) {
                if (isupper(word[i])) {
                    return false;
                }
            }
        }
        return true;
    }
};
